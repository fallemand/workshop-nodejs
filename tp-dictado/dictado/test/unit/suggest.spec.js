const React = require('react');
const renderer = require('react-test-renderer');
const Suggest = require('../../views/components/Suggest');

describe('Suggest (Snapshot)', () => {
  const component = renderer.create(<Suggest query="iph"/>);
  const suggestResponse = {
    "results": [
      "iphone 6",
      "iphone 7",
      "iphone 5",
    ],
  };

  afterEach(() => {
    // Clean component state
    const instance = component.getInstance();
    instance.setState({
      results: []
    });
  });

  test('Suggest renders resultssuggestResponse [change state]', () => {
    const instance = component.getInstance();
    instance.setState(suggestResponse);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('Suggest renders without results', () => {
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('Suggest renders results [call componentWillReceiveProps]', () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({json: () => Promise.resolve(suggestResponse)}));
    expect.assertions(1);
    const instance = component.getInstance();
    return instance.componentWillReceiveProps({query: 'iph'}).then(() => {
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });
});
