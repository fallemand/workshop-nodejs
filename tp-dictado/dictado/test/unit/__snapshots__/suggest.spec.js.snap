// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggest (Snapshot) Suggest renders results [call componentWillReceiveProps] 1`] = `
<div>
  <ul
    className="suggest"
  >
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=iphone 6"
      >
        iphone 6
      </a>
    </li>
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=iphone 7"
      >
        iphone 7
      </a>
    </li>
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=iphone 5"
      >
        iphone 5
      </a>
    </li>
  </ul>
</div>
`;

exports[`Suggest (Snapshot) Suggest renders resultssuggestResponse [change state] 1`] = `
<div>
  <ul
    className="suggest"
  >
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=iphone 6"
      >
        iphone 6
      </a>
    </li>
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=iphone 7"
      >
        iphone 7
      </a>
    </li>
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=iphone 5"
      >
        iphone 5
      </a>
    </li>
  </ul>
</div>
`;

exports[`Suggest (Snapshot) Suggest renders without results 1`] = `<div />`;
